packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -625,37 +640,57 @@ class _RootBuilder extends dart_ast_visitor.RecursiveAstVisitor<Object?> {
         );
       }
     } else {
-      _currentClass = Class(name: node.name.name, fields: <Field>[]);
+      _currentClass = Class(name: node.name.name, fields: <NamedType>[]);
     }
 
     node.visitChildren(this);
     return null;
   }
 
+  NamedType formalParameterToField(dart_ast.FormalParameter parameter) {
+    final dart_ast.TypeName typeName = parameter.childEntities.firstWhere(
+        (dart_ast_syntactic_entity.SyntacticEntity e) =>
+            e is dart_ast.TypeName) as dart_ast.TypeName;
+    final String argType = typeName.name.name;
```

Comment:
I thought when I said that `dataType` was a confusing name for the base type given that "data type" is the whole thing, you said this was the Dart AST name. But here it's "name", which was one of my suggestions for what to call it. Can we revisit that?

(If the only reason you don't want to call it "name" is because of the collision with `NamedType`'s name, as I said before `TypedEntity` could easily remap `name` to `dataType` in its getter.)

Reply:
> you said this was the Dart AST name
I didn't intend to convey that.  `typeName.name.name` is pretty ugly fwiw, we shouldn't copy that haha.

I think ` field.type.dataType + ' ' + field.name` is more clear than `field.type.name + ' ' + field.name` considering we are consistent about what `dataType` means across the codebase.  `.name` is easier when reading the code for the first time, but `.dataType` is easier when maintaining the code since it is codifying context into its name.   Like, read the code as are suggesting then answer the question what is a "name"?  The answer is "it depends on the context".  Read the code like it is now and answer the question "what is a dataType"? The answer is "the name of a type devoid of the information that makes it a functional type (ie nullability and type arguments)".

I think `.baseName` is better than `.name`, but I prefer '.dataType`.  Do you like `.baseName`?

Reply:
> I think ` field.type.dataType + ' ' + field.name` is more clear than `field.type.name + ' ' + field.name` considering we are consistent about what `dataType` means across the codebase.

It is consistent, but borderline meaningless because it's just a slightly different phrasing of the name of the class it's part of. One could make the same argument for `field.variableField` instead of `field.name`.

> `.name` is easier when reading the code for the first time, but `.dataType` is easier when maintaining the code since it is codifying context into its name.

The context of things in classes is always the class. Are you suggesting that it's good practice to put the name of the class in the name of every public element of a class so that it can be disambiguated without looking at the class?

> Like, read the code as are suggesting then answer the question what is a "name"? The answer is "it depends on the context".

The context that is literally immediately before the token "name", yes, because that's how scoped access works.

Are you saying that someone reading `field.type.name` is not going to understand that it's the name of the field's type without going and looking up some external context, even though that's exactly the plain English reading of that code?

> Read the code like it is now and answer the question "what is a dataType"? The answer is "the name of a type devoid of the information that makes it a functional type (ie nullability and type arguments)".

If you've gone to look up the documentation of `dataType`, because nothing about the name communicates that information. And it is no easier to find the documentation of `type.dataType` than `type.name`, because, again, the context is literally right there.

> I think `.baseName` is better than `.name`, but I prefer `.dataType`. Do you like `.baseName`?

Yes, since I think it is at least trying to communicate what element of the type it is, rather than just sounding like it's the whole type.

Reply:
Done, renamed it to "baseName"